Backend/app/rider-platform/rider-app/Main/src/Beckn/ACL/FRFS/Cancel.hs

```diff
@@ -31,32 +31,33 @@ buildCancelReq ::
   DBooking.FRFSTicketBooking ->
   BecknConfig ->
   Utils.BppData ->
+  Spec.CancellationType ->
   m (Spec.CancelReq)
-buildCancelReq booking bapConfig bppData = do
+buildCancelReq booking bapConfig bppData cancellationType = do
   now <- getCurrentTime
   let transactionId = booking.searchId.getId
       validTill = addUTCTime (intToNominalDiffTime 30) now
       ttl = diffUTCTime validTill now
-  messageId <- generateGUID
+      messageId = booking.id.getId
```

Comment:
this will make `messageId` same for both cancellation types, which shouldn't be the case.
You can lookup booking based on `transaction_id`